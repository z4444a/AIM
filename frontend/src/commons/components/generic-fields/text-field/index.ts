import { TextField } from './text-field';
import { withTranslation } from 'react-i18next';

export default withTranslation('common')(TextField);
