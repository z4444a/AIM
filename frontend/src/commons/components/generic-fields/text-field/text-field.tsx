import React, { ReactNode } from 'react';
import MUITextField from '@material-ui/core/TextField';
import {
  ErrorConfig,
  TextFieldConfigModel,
  TextFieldHandlers,
  TextFieldValue,
} from '../models/field-config.model';
import Tooltip from '@material-ui/core/Tooltip';
import AimMultipleInput from '../../aim-multiple-input/index';
import parameterConstraintsService from '../../../services/parameter-constraints.service';
import { WithTranslation } from 'react-i18next';

export interface Props {
  fieldConfig: TextFieldConfigModel;
  handlers: TextFieldHandlers;
  value?: TextFieldValue;
  values?: TextFieldValue[];
  needTooltip?: boolean;
}

export type InternalProps = Props & WithTranslation;
interface State {
  inputValue: string;
  localConfig: TextFieldConfigModel;
}
export class TextField extends React.PureComponent<InternalProps, State> {
  private readonly emptyValue = '';
  constructor(props: InternalProps) {
    super(props);
    this.state = {
      inputValue: this.emptyValue,
      localConfig: props.fieldConfig,
    };
  }

  componentDidUpdate(prevProps: Readonly<InternalProps>, prevState: Readonly<State>): void {
    this.setLocalConfig(prevState.localConfig.error);
  }

  public render(): ReactNode {
    const { value, fieldConfig, needTooltip, values } = this.props;
    const { error, className, style, required, labelConfig, multipleMax } = fieldConfig;
    const multiple = (multipleMax && multipleMax > 1) as boolean;
    const singleValue = value ? value : this.emptyValue;
    const { localConfig, inputValue } = this.state;
    const tooltipTitle = needTooltip
      ? this.getTooltip(
          fieldConfig.constraints ? fieldConfig.constraints.maxStringLength : undefined
        )
      : '';
    return multiple ? (
      <Tooltip title={tooltipTitle}>
        <div>
          <AimMultipleInput
            fieldConfig={localConfig}
            values={values ? values : []}
            onValuesChange={this.handleValuesChange}
            inputValue={inputValue}
            onInputChange={(input: string) => this.setState({ inputValue: input })}
          />
        </div>
      </Tooltip>
    ) : (
      <Tooltip title={tooltipTitle}>
        <MUITextField
          style={style}
          className={className}
          error={!!error || !!localConfig.error}
          helperText={localConfig.error ? localConfig.error.value : undefined}
          label={labelConfig && labelConfig.visible ? labelConfig.value : undefined}
          required={required}
          value={singleValue}
          onChange={this.handleValueChange}
        />
      </Tooltip>
    );
  }
  private setLocalConfig = (prevError?: ErrorConfig) => {
    const { fieldConfig, value } = this.props;
    const multiple = fieldConfig.multipleMax && fieldConfig.multipleMax > 1;
    const { error, ...rest } = fieldConfig;
    const { inputValue } = this.state;
    parameterConstraintsService.setI18n({ ...this.props });
    const localErrorModel = parameterConstraintsService.checkTextConstraints(
      fieldConfig,
      multiple ? (inputValue as TextFieldValue) : value
    );
    const localError = localErrorModel ? localErrorModel : undefined;
    const newError = error ? error : localError;
    if ((prevError && !newError) || (newError && !prevError)) {
      this.setState({ localConfig: { ...rest, error: newError } });
    }
  };

  private handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { handlers, fieldConfig } = this.props;
    if (!handlers || typeof handlers.onValueChange !== 'function') {
      return;
    }
    handlers.onValueChange(event.target.value, fieldConfig.key);
  };

  private handleValuesChange = (values: string[]) => {
    const { handlers, fieldConfig } = this.props;
    const { key } = fieldConfig;
    if (!handlers || typeof handlers.onValuesChange !== 'function') {
      return;
    }
    handlers.onValuesChange(values, key);
  };

  private getTooltip = (length: number | undefined): string => {
    return 'Введите текст' + (length ? ' не длиннее ' + length + ' символов' : '');
  };
}
