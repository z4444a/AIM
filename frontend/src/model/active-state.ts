export const ActiveState = [
  {
    key: '1',
    value: 'Активен',
  },
  {
    key: '2',
    value: 'Не активен',
  },
];
