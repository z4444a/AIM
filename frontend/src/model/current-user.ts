import { Role } from '../commons/role';

export interface CurrentUser {
  id: number;
  fullName: string;
  username: string;
  role: Role;
}
