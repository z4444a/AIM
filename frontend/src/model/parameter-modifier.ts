export enum ParameterModifier {
  POOL_PARAMETER = 1,
  REQUEST_PARAMETER = 2,
  ALLOCATION_PARAMETER = 3,
}
