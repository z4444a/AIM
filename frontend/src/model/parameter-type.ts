export enum ParameterType {
  NUMBER = 'NUMBER',
  TEXT = 'TEXT',
  DATE = 'DATE',
  LIST = 'LIST',
  REAL = 'REAL',
  POOL = 'POOL',
}
