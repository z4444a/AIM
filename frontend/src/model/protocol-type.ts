export const Protocols = [
  {
    key: '1',
    value: 'TLS',
  },
  {
    key: '2',
    value: 'SSL',
  },
  {
    key: '3',
    value: 'None',
  },
];
