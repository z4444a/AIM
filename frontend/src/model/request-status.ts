import { RequestStatus } from './get/request-status';

export function RequestStatuses(t: (s: string) => string) {
  return [
    {
      key: RequestStatus.NEW.toString(),
      value: t('requestPage.status.new'),
    },
    {
      key: RequestStatus.IN_PROGRESS.toString(),
      value: t('requestPage.status.inWork'),
    },
    {
      key: RequestStatus.PROCESSED.toString(),
      value: t('requestPage.status.executed'),
    },
    {
      key: RequestStatus.CANCELED.toString(),
      value: t('requestPage.status.canceled'),
    },
    {
      key: RequestStatus.PAUSED.toString(),
      value: t('requestPage.status.paused'),
    },
  ];
}
