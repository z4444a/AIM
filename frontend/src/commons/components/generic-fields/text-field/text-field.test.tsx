import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { TextFieldConfigModel, TextFieldHandlers } from '../models/field-config.model';
import { InternalProps, TextField } from './text-field';

describe('TextField', () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    const fieldConfig = new TextFieldConfigModel('textField');
    const handlers: TextFieldHandlers = {
      onValueChange: () => {},
    };

    const props: InternalProps = {
      fieldConfig,
      handlers,
    };
    wrapper = shallow(<TextField {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
