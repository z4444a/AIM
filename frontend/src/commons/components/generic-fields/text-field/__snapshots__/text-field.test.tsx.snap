// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField should render correctly 1`] = `
<WithStyles(Tooltip)
  title=""
>
  <TextField
    error={false}
    onChange={[Function]}
    required={false}
    select={false}
    value=""
    variant="standard"
  />
</WithStyles(Tooltip)>
`;
